packages/mobile/src/identity/feelessVerification.ts

```diff
@@ -0,0 +1,1061 @@
+  AttestationCode,
+  attestationCodeReceiver,
+  completeAttestations,
+  getActionableAttestations,
+  getAttestationsStatus,
+  NUM_ATTESTATIONS_REQUIRED,
+  requestAndRetrieveAttestations,
+  revealAttestations,
+  VERIFICATION_TIMEOUT,
+} from 'src/identity/verification'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import Logger from 'src/utils/Logger'
+import { setMtwAddress } from 'src/web3/actions'
+import { getContractKit } from 'src/web3/contracts'
+import { registerWalletAndDekViaKomenci } from 'src/web3/dataEncryptionKey'
+import { getConnectedAccount, getConnectedUnlockedAccount } from 'src/web3/saga'
+import { TransactionReceipt } from 'web3-eth'
+
+const TAG = 'identity/feelessVerification'
+
+const KOMENCI_URL = 'https://komenci.celo-networks-dev.org'
+// const KOMENCI_URL = 'http://192.168.86.33:3000'
+// TODO: Populate this with expected implementation address
+const ALLOWED_MTW_IMPLEMENTATIONS: Address[] = ['0x88a2b9B8387A1823D821E406b4e951337fa1D46D']
+const CURRENT_MTW_IMPLEMENTATION_ADDRESS: Address = '0x88a2b9B8387A1823D821E406b4e951337fa1D46D'
+
+export function* feelessFetchVerificationState() {
+  Logger.debug(TAG, '@feelessFetchVerificationState', 'Starting fetch')
+  try {
+    ValoraAnalytics.track(VerificationEvents.verification_fetch_status_start)
+    const [contractKit, walletAddress, feelessVerificationState, e164Number]: [
+      ContractKit,
+      string,
+      FeelessVerificationState,
+      string
+    ] = yield all([
+      call(getContractKit),
+      call(getConnectedUnlockedAccount),
+      select(feelessVerificationStateSelector),
+      select(e164NumberSelector),
+      put(feelessSetVerificationStatus(VerificationStatus.GettingStatus)),
+    ])
+
+    const komenciKit = new KomenciKit(contractKit, walletAddress, {
+      url: KOMENCI_URL,
+      token: feelessVerificationState.komenci.sessionToken,
+    })
+
+    try {
+      // Checks if a MTW was verified in a previous attempt and updates state.
+      // This check will likely fail because the pepper hasn't been cached yet
+      // but needs to happen for the edge case that a user has a cached pepper
+      // and Komenci is down
+      yield call(fetchVerifiedMtw, contractKit, walletAddress, e164Number)
+    } catch (error) {
+      Logger.debug(TAG, 'Unable to check it MTW is verified on first attempt')
+    }
+    // Throws error if Komenci is not ready for user, otherwise it updates state
+    yield call(fetchKomenciReadiness, komenciKit)
+
+    // Throws error if unable to retreive phone hash, otherwise it updates state
+    yield call(fetchPhoneHashDetailsFromCache, e164Number)
+
+    // Updates state with sessionStatus and the mtwAddress associated with the session
+    yield call(fetchSessionState, komenciKit, e164Number)
+
+    // Checks if a MTW was verified in a previous attempt and updates state
+    yield call(fetchVerifiedMtw, contractKit, walletAddress, e164Number)
```

Comment:
At this point, we would only have a `unverifiedMtwAddress` if the user had an active Komenci session they were resuming. What about users that already verified but had their Komenci session expire?

Reply:
One the things I was confused about here was why, if the `fetchVerifiedMtw` call above failed, this one would ever succeed? IIUC correctly we don't fetch the pepper in this flow, so if it wasn't cached before, we still will not have it now. More generally, I understand the case of why we want to call `fetchVerifiedMtw` in this flow, but don't understand why we do it twice. It would be better to find some way to check at most once.

Reply:
Yeah, you are right. I can add logic so that this check only happens once. If Komenci is live, it will happen at the end of `feelessFetchVerificationState`. If Komenci is down, it will check it once before bailing out on the rest of the checks. But there is no reason why it should ever happen twice.